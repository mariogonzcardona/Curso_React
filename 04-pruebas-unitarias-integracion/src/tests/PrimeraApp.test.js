import React from "react";
import '@testing-library/jest-dom';
import {shallow} from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en PrimeraApp.js", () => {

    // test('Debe de mostrar el mensaje Hola, soy Mario', () => {
    //     const saludo="Hola, soy Mario";
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });

    test('Debe de mostrar <PrimeraApp/> correctamente', () => {
        const saludo="Hola, soy Mario";
        const wrapper=shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo mostrado por props', () => {
        const saludo="Hola, soy Mario";
        const subtitulo="Soy un subtitulo";

        const wrapper=shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );
        const textoParrafo=wrapper.find('p').text().trim();
        expect(textoParrafo).toBe(subtitulo);
    });
});