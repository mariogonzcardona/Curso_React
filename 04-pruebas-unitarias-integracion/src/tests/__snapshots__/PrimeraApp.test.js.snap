// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp.js Debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Mario
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
